fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/domain/SearchQueryRenderer.java

```diff
@@ -0,0 +1,2042 @@
+    protected WhereFragment paramFilter(QueryParameter queryParm, String paramTableAlias) throws FHIRPersistenceException {
+        final WhereFragment result;
+
+        switch (queryParm.getType()) {
+        case URI:
+        case STRING:
+            result = getStringFilter(queryParm, paramTableAlias);
+            break;
+        case NUMBER:
+            result = getNumberFilter(queryParm, paramTableAlias);
+            break;
+        case QUANTITY:
+            result = getQuantityFilter(queryParm, paramTableAlias);
+            break;
+        case DATE:
+            result = getDateFilter(queryParm, paramTableAlias);
+            break;
+        case SPECIAL:
+            result = getLocationFilter(queryParm, paramTableAlias);
+            break;
+        case REFERENCE:
+            result = getReferenceFilter(queryParm, paramTableAlias);
+            break;
+        case TOKEN:
+            result = getTokenFilter(queryParm, paramTableAlias);
+            break;
+        default:
+            result = null;
+            break;
+        }
+
+        if (result == null) {
+            throw new FHIRPersistenceException("Nested composite parameters are not supported");
+        }
+
+        return result;
+    }
+
+    @Override
+    public QueryData addLocationPosition(QueryData queryData, List<QueryParameter> queryParameters) throws FHIRPersistenceException {
+        // Special handling for location position extension logic
+        NearLocationHandler handler = new NearLocationHandler();
+        List<Bounding> boundingAreas;
+        try {
+            boundingAreas = handler.generateLocationPositionsFromParameters(queryParameters);
+        } catch (FHIRSearchException e) {
+            throw new FHIRPersistenceException("input parameter is invalid bounding area, bad prefix, or bad units", e);
+        }
+
+        if (!boundingAreas.isEmpty()) {
+            buildLocationQuerySegment(queryData, boundingAreas);
+        }
+
+        return null;
+    }
+
+    /**
+     * Extension filter not associated with a particular parameter?
+     * @param queryData
+     * @param boundingAreas
+     */
+    protected void buildLocationQuerySegment(QueryData queryData, List<Bounding> boundingAreas) {
+//        final String parameterName = queryParm.getCode();
+//        final int aliasIndex = getNextAliasIndex();
+//        final SelectAdapter query = queryData.getQuery();
+//        final String paramTableName = resourceType + "_LATLNG_VALUES";
+//        final String paramAlias = "P" + aliasIndex;
+//        final String lrAlias = queryData.getLRAlias();
+//        ExpNode filter = getLocationFilter(queryParm, paramAlias).getExpression();
+//        query.from().innerJoin(paramTableName, alias(paramAlias), on(paramAlias, "LOGICAL_RESOURCE_ID").eq(lrAlias, "LOGICAL_RESOURCE_ID")
+//            .and(paramAlias, "PARAMETER_NAME_ID").eq(getParameterNameId(parameterName))
+//            .and(filter));
+//
+//        return queryData;
+    }
```

Comment:
What is this?

Reply:
old code no longer required (based on unit-test and system integration test passes. As our "Location" expert, I think having your eyes on the new location search implementation would be helpful, in case there are gaps not covered by our tests.

Reply:
I removed the above method, and added a FINE log statement.

Reply:
I spent about a half hour going through the NewLocationBehavior... I'm pretty sure it's good

Reply:
I spent about a half hour going through the NewLocationBehavior... I'm pretty sure it's good